code/game/objects/items/manuals.dm

```diff
@@ -1013,4 +1022,213 @@
 /datum/action/innate/drink_fling/Deactivate()
 	button_icon_state = "drinkfling_off"
 	active = FALSE
-	UpdateButtonIcon()
\ No newline at end of file
+	UpdateButtonIcon()
+
+/obj/item/book/granter/spell
+	var/spell = null
+	var/spellname = "magical bugs"
+	var/oneuse = TRUE //default this is one, but admins can var this to 0 if we wanna all have a pass around of the rod form book
+	var/used = FALSE //only really matters if oneuse but it might be nice to know if someone's taken
+
+/obj/item/book/granter/spell/attack_self(mob/user)
+	if(!spell)
+		return
+	if(reading)
+		to_chat(user, "<span class='warning'>You're already reading this!</span>")
+		return
+	var/obj/effect/proc_holder/spell/S = new spell
+	for(var/obj/effect/proc_holder/spell/knownspell in user.mind.spell_list)
+		if(knownspell.type == S.type)
+			if(user.mind)
+				if(iswizard(user))
+					to_chat(user,"<span class='notice'>You're already far more versed in this spell than this flimsy how-to book can provide.</span>")
+				else
+					to_chat(user,"<span class='notice'>You've already read this one.</span>")
+			return
+	if(used && oneuse)
+		recoil(user)
+	else
+		to_chat(user, "<span class='notice'>You start reading about casting [spellname]...</span>")
+		reading = TRUE
+		for(var/i=1, i<=pages_to_mastery, i++)
+			if(!turn_page(user))
+				to_chat(user, "<span class='notice'>You stop reading...</span>")
+				reading = FALSE
+				qdel(S)
+				return
+		if(do_after(user,50, user))
+			to_chat(user, "<span class='notice'>You feel like you've experienced enough to cast [spellname]!</span>")
+			reading = FALSE
+			user.mind.AddSpell(S)
+			user.log_message("<font color='orange'>learned the spell [spellname] ([S]).</font>", INDIVIDUAL_ATTACK_LOG)
+			onlearned(user)
+
+/obj/item/book/granter/spell/proc/recoil(mob/user)
+	user.visible_message("<span class='warning'>[src] glows in a black light!</span>")
+
+/obj/item/book/granter/spell/proc/onlearned(mob/user)
+	used = TRUE //might seem odd to not put this under the oneuse check but if many people use it off oneuse and then an admin sets it to oneuse i could make an argument for having it then immediately jump to recoil
+	if(oneuse)
+		user.visible_message("<span class='caution'>[src] glows dark for a second!</span>")
+
+/obj/item/book/granter/spell/fireball
+	spell = /obj/effect/proc_holder/spell/aimed/fireball
+	spellname = "fireball"
+	icon_state ="bookfireball"
+	desc = "This book feels warm to the touch."
+	remarks = list("Aim...AIM, FOOL!", "Just catching them on fire won't do...", "Accounting for crosswinds... really?", "I think I just burned my hand...", "Why the dumb stance? It's just a flick of the hand...", "OMEE... ONI... Ugh...", "What's the difference between a fireball and a pyroblast...")
+
+/obj/item/book/granter/spell/fireball/recoil(mob/user)
+	..()
+	explosion(user.loc, -1, 0, 2, 3, 0, flame_range = 2)

```

Comment:
explosion(user.loc, -1, 0, 2, 3, FALSE, FALSE, 2)

Reply:
though that said, is -1 devastation_range intended?

Reply:
Just doing what the old spell book did let's not do balance yes?